xmodule/docs/decisions/0003-library-content-block-schema.rst

```diff
@@ -0,0 +1,201 @@
+| **shuffle = True,  max_count = -1** | Entire library in random order (V1's "random mode")           | Manual subset of library in random order                        |
++-------------------------------------+---------------------------------------------------------------+-----------------------------------------------------------------+
+| **shuffle = True,  max_count > 0**  | Random subset of library in random order (V1's "random mode") | Random subset of manual subset of library in random order       |
++-------------------------------------+---------------------------------------------------------------+-----------------------------------------------------------------+
+| **shuffle = False, max_count = -1** | Entire library in original order                              | Manual subset of library in original order ("static mode")      |
++-------------------------------------+---------------------------------------------------------------+-----------------------------------------------------------------+
+| **shuffle = False, max_count > 0**  | Random subset of library in original order                    | Random subset of manual subset of library in original order     |
++-------------------------------------+---------------------------------------------------------------+-----------------------------------------------------------------+
+
+At first, we will only aim to support the "random mode" behaviors plus the new "static mode" behavior. Validation will be used to ensure that the other modes are not available. In the future, we could expect to loosen this restriction.
+
+The final library_content block schema, with all changes, will look like this:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Field Name
+     - Scope, Type
+     - Default
+     - Example
+     - Description
+
+   * - source_library_id
+     - settings, str|None
+     - None
+     - "lib:Open-edX:ExampleLib"
+     - Key of V1 or V2 source library; None means unselected.
+
+   * - source_library\_ version
+     - settings, str|None
+     - None
+     - TBD
+     - Version of V1 source library (MongoID string) or V2 source library (a stringified int). None means latest.
+
+   * - manual
+     - settings, bool
+     - False
+     - True
+     - When False, all library blocks matching capa_type are copied as library_content children, including newly-added library blocks when upgrading source library version. When True, the course author is propmted to pick specific blocks from the library; these blocks become the library_content children. Studio respects these manual block choices (i.e., it won't auto-add new library blocks when the library version is updated).
+
+   * - capa_type
+     - settings, str
+     - "any"
+     - "symbolicresponse"
+     - Unless "any": Include only problems, and only problems of this response type. Setting manual to True overrides this filter, however it could still be used for filtering in the block-picker UI.
```

Comment:
FWIW, I realize this is a backwards compatibility thing, but having a more block-agnostic selection criteria in the future would be nice. :-/

Reply:
Oh definitely. I would love to see this field eventually deprecated in favor of, like, `block_type` and/or `tags`.